x-pack/plugins/cases/public/containers/use_get_feature_ids.tsx

```diff
@@ -6,28 +6,66 @@
  */
 
 import { useQuery } from '@tanstack/react-query';
-import type { ValidFeatureId } from '@kbn/rule-data-utils';
+import { isValidFeatureId } from '@kbn/rule-data-utils';
+import { useEffect, useMemo, useState } from 'react';
 import type { ServerError } from '../types';
 import { useCasesToast } from '../common/use_cases_toast';
 import * as i18n from './translations';
+import type { FeatureIdsResponse } from './api';
 import { getFeatureIds } from './api';
 import { casesQueriesKeys } from './constants';
 
-export const useGetFeatureIds = (alertRegistrationContexts: string[]) => {
+export const useGetFeatureIds = (
+  alertIds: string[],
+  alertIdsQuery: {
+    ids: {
+      values: string[];
+    };
+  },
+  enabled: boolean
+) => {
   const { showErrorToast } = useCasesToast();
-
-  return useQuery<ValidFeatureId[], ServerError>(
-    casesQueriesKeys.alertFeatureIds(alertRegistrationContexts),
+  const [featureIds, setFeatureIds] = useState<string[]>([]);
+  const { data, isSuccess, isInitialLoading, isLoading } = useQuery<
+    FeatureIdsResponse,
+    ServerError
+  >(
+    casesQueriesKeys.alertFeatureIds(alertIds),
     ({ signal }) => {
-      const query = { registrationContext: alertRegistrationContexts };
-      return getFeatureIds({ query, signal });
+      return getFeatureIds({ query: alertIdsQuery, signal });
     },
     {
+      enabled,
       onError: (error: ServerError) => {
         showErrorToast(error, { title: i18n.ERROR_TITLE });
       },
     }
   );
+
+  useEffect(() => {
+    if (isSuccess && data) {
+      const localFeatureIds = new Set<string>();
+      data?.aggregations?.consumer?.buckets?.forEach(
+        ({ key, doc_count: docCount }: { key: string; doc_count: number }) => {
+          if (docCount > 0 && isValidFeatureId(key)) {
+            localFeatureIds.add(key);
+          }
+        }
+      );
+      data?.aggregations?.producer?.buckets?.forEach(
+        ({ key, doc_count: docCount }: { key: string; doc_count: number }) => {
+          if (docCount > 0 && isValidFeatureId(key)) {
+            localFeatureIds.add(key);
+          }
+        }
+      );
+      setFeatureIds([...localFeatureIds]);
+    }
+  }, [data, isSuccess]);
```

Comment:
`setFeatureIds` is only used in this effect. should we replace assigning of the `featureIds` with `useMemo` instead?

Reply:
Yes avoiding a state might be better but at the end same # of rendering.